src/fuddly/framework/plumbing.py

```diff
@@ -759,125 +758,159 @@ def monitor_probes(self, prefix=None, force_record=False):
+                else:
+                    self.import_successfull = False
 
+    def _get_data_models_from_modules(self, fmkDB_update=True):
         if not self._quiet:
-            self.print(colorize(FontStyle.BOLD + "=" * 63 + "[ Data Models ]==", rgb=Color.FMKINFOGROUP))
+            self.print(colorize(FontStyle.BOLD + "="*63+"[ Data Models (python modules) ]==", rgb=Color.FMKINFOGROUP))
 
-        for dname, file_list in data_models.items():
-            if not self._quiet:
-                self.print(colorize(">>> Look for Data Models within '%s' directory" % dname,
-                               rgb=Color.FMKINFOSUBGROUP))
-            prefix = dname.replace(os.sep, ".") + "."
-            for f in file_list:
-                res = rexp_strategy.match(f)
-                if res is None:
-                    continue
-                name = res.group(1)
-                if name + ".py" in file_list:
-                    dm_params = self._import_dm(prefix, name)
-                    if dm_params is not None:
-                        self._add_data_model(
-                            dm_params["dm"],
-                            dm_params["tactics"],
-                            dm_params["dm_rld_args"],
-                            reload_dm=False,
-                        )
-                        self.__dyngenerators_created[dm_params["dm"]] = False
-                        if fmkDB_update:
-                            # populate FMK DB
-                            self._fmkDB_insert_dm_and_dmakers(dm_params["dm"].name, dm_params["tactics"])
-                    else:
-                        self.import_successfull = False
+        group_name=gr.ep_group_names["data_models"]
+        dms = entry_points(group=group_name)
+        for module in dms:
+            try:
+                dm_params = self._import_dm(module, module.name)
+            except DataModelDuplicateError as e:
+                if not self._quiet:
+                    self.print(colorize(f"*** The data model '{e.name}' was already defined, "
+                                        f"ignoring... [{module.module}] ***", rgb=Color.WARNING))
+                continue
 
-        if fmkDB_update:
-            self.fmkDB.insert_data_model(Database.DEFAULT_DM_NAME)
-            self.fmkDB.insert_dmaker(Database.DEFAULT_DM_NAME, Database.DEFAULT_GTYPE_NAME,
-                                     Database.DEFAULT_GEN_NAME, True, True)
+            if dm_params is not None:
+                self._add_data_model(dm_params['dm'], dm_params['tactics'], dm_params['dm_rld_args'], 
+                                     reload_dm=False)
+                name = dm_params['dm'].name
+                self.__dyngenerators_created[dm_params['dm']] = False
+                if fmkDB_update:
+                    # populate FMK DB
+                    self._fmkDB_insert_dm_and_dmakers(dm_params['dm'].name, dm_params['tactics'])
+            else:
```

Comment:
same as before, if None: =false; break

Reply:
Same as before, fmkDB_update is a boolean